### PR TITLE
Transition from `JSON::Matchers` to `JsonMatchers`

### PR DESCRIPTION
**Pull request by seanpdoyle**
**BREAKING CHANGE**

Rename module to `JsonMatchers`. This resolves clashing with gems like
`oj` / `oj_mimic_json` that take control of the standard library's
`json` module. As a result, the file to require is now `json_matchers`,
instead of `json/matchers`.

This should leave clients largely unaffected, since their usage of the
gem is based around `require "json/matchers"` and using a globally
accessible `match_response_schema`.

The only real difference is that the gem name (`json-matchers`) is no
longer aligned with Ruby conventions around module nesting. As a
consumer, I would expect the gem name to be `json_matchers`.

For backwards compatibility, this PR introduces the `json/matchers` file
as an alias for `json_matchers`, so that consumers can still call
`require "json/matchers"`.

For compatibility with the gem name, consumers can also call `require
"json-matchers"`, which is aliased to `require "json_matchers"`.

No longer condone auto-loading RSpec. Add documentation around adding
`require "json_matchers/rspec"` to consumers' `spec/spec_helper.rb`

### CODE REVIEW

**Comment by seanpdoyle** [comment]
@mike-burns would it be worthwhile to rename:

* This repo to `json_matchers`
* The gem to `json_matchers`, maybe pushing a `json-matchers@0.4.0` that outputs a deprecation warning?

**Comment by mike-burns** [comment]
I've no complaints about this, but maybe someone else has stronger opinions.

**Comment by seanpdoyle** [comment]
@mike-burns thanks. Do you have strong opinions about the gem's name vs its module structure?

That is to say `json-matchers` -> `JSON::Matchers` vs `json_matchers` -> `JsonMatchers`?

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -1,3 +1,5 @@
+require "json-matchers"
+require "json_matchers"
 require "json/matchers"
```
Comment (seanpdoyle) [inline spec/spec_helper.rb]: This ensures that all flavors of the require are tested.

If they're working properly, the first should evaluate, while the rest return `false` and no-op.

Is there a better way?

**Comment by teoljungberg** [comment]
Aslong as there's a fallback for the require, I think we're good

**Comment by tabfugnic** [comment]
It still might be good to have some transition release such as a 4.0 release that adds deprecation warnings.

**Comment by tabfugnic** [comment]
I don't feel that strongly about it though

**Comment by seanpdoyle** [comment]
What if we output deprecation warnings in `lib/json/matchers.rb`?

**Comment by tabfugnic** [comment]
:+1:

**Pull request merged**